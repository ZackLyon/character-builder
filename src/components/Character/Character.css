.character {
  display: flex;
  flex-direction: column;
}

.head,
.middle,
.legs {
  width: 200px;

  background-size: contain;
  background-repeat: no-repeat;
}

.head {
  height: 200px;
  margin-bottom: -5px;
}

.middle,
.legs {
  height: 300px;
}

.legs {
  margin-top: -60px;
}
