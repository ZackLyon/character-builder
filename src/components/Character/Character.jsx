import React from 'react';
import './Character.css';

export default function Character({ head, middle }) {
  return (
    <div className='character'>
      <div
        className='head'
        style={{ backgroundImage: `url(${head}-head.png)` }}
      ></div>
      <div
        className='middle'
        style={{ backgroundImage: `url(${middle}-middle.png)` }}
      ></div>
    </div>
  );
}
