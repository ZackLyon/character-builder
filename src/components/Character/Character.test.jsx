import { render } from '@testing-library/react';
import Character from './Character';

it('should display the character on the screen', () => {
  const { container } = render(
    <Character head='angry' middle='blue' leg='dog' />
  );

  expect(container).toMatchSnapshot();
});
