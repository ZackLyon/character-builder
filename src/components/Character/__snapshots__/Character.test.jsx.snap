// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the character on the screen 1`] = `
<div>
  <div
    class="character"
  >
    <div
      class="head"
      style="background-image: url(angry-head.png);"
    />
    <div
      class="middle"
      style="background-image: url(blue-middle.png);"
    />
    <div
      class="legs"
      style="background-image: url(dog-pants.png);"
    />
  </div>
</div>
`;
